lib/post/post.controllers.js

```diff
@@ -0,0 +1,157 @@
+const Joi = require('joi');
+
+module.exports = (models) => {
+  const Boom = require('boom');
+  const Sequelize = require('sequelize');
+  const Op = Sequelize.Op;
+  const postModels = require('./post.models');
+  const accountLib = require('../account/account.controllers')(models).lib;
+  const groupLib = require('../group/group.controllers')(models).lib;
+
+  return {
+    getPosts: async function(request, h) {
+      let returnPosts;
+      try {
+        returnPosts = await getPosts(request.query);
+      } catch (e) {
+        throw Boom.badImplementation('Error during getPosts(). ' + e);
+      }
+      return h
+        .response(returnPosts);
+    },
+
+    getPost: async function(request, h) {
+      const returnPost = await getPost(request.params.id);
+      // TODO: use try/catch with await?
+      return h
+        .response(returnPost);
+    },
+
+    postPost: async function (request, h) {
+      let groupId = request.params.id;
+      let account;
+
+      try {
+        account = await accountLib.getAccount(request.auth.credentials);
+      } catch (e) {
+        throw Boom.badImplementation('Error during getAccount(request.auth.credentials). ' + e);
+      }
+
+      if (account === null) {
+        throw Boom.badRequest('Must have an Account to create a Comment');
+      }
+
+      let groupItem;
+      try {
+        groupItem = await getGroupItem(groupId);
```

Comment:
this is overcomplicated in the same way as getPostItem. Please correct in similar fashion.